gauntlet/packages/gauntlet-solana/src/lib/feeUtils.ts

```diff
@@ -0,0 +1,90 @@
+import { PublicKey, BlockResponse, VOTE_PROGRAM_ID } from '@solana/web3.js'
+import { bs58 } from '@coral-xyz/anchor/dist/cjs/utils/bytes'
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+
+export interface BlockData {
+  fees: number[]
+  prices: number[]
+}
+
+const COMPUTE_BUDGET_PROGRAM_ID = new PublicKey('ComputeBudget111111111111111111111111111111')
+
+// Instructions are serialized in order in rust
+// https://github.com/solana-labs/solana/blob/60858d043ca612334de300805d93ea3014e8ab37/sdk/src/compute_budget.rs#L25
+enum ComputeBudgetInstruction {
+  SetComputeUnitLimit = 0x02,
+  SetComputeUnitPrice = 0x03,
+}
+
+export const parseBlock = (block: BlockResponse): BlockData => {
+  const blockData: BlockData = { fees: [], prices: [] }
+  for (const tx of block.transactions) {
+    const meta = tx.meta
+    let computeUnitPrice: number
+    // filter out consensus vote transactions
+    // consensus messages are included as txs within blocks
+    // validate AccountKeys has enough elements to index into ProgramIDIndex
+    if (
+      tx.transaction.message.instructions.length == 1 &&
+      tx.transaction.message.accountKeys.length > tx.transaction.message.instructions[0].programIdIndex &&
+      tx.transaction.message.accountKeys[tx.transaction.message.instructions[0].programIdIndex] == VOTE_PROGRAM_ID
+    ) {
+      continue
+    }
+
+    for (const ins of tx.transaction.message.instructions) {
+      const programId = tx.transaction.message.accountKeys[ins.programIdIndex]
+
+      // Check if instruction is from Compute Budget Program
+      if (programId.equals(COMPUTE_BUDGET_PROGRAM_ID)) {
+        const parsedPrice = parseComputeUnitPrice(Buffer.from(bs58.decode(ins.data)))
+        if (parsedPrice !== null) {
+          computeUnitPrice = parsedPrice
+          break // Only one SetComputeUnitPrice instruction per transaction
+        }
+      }
+    }
+
+    if (meta && computeUnitPrice) {
```

Comment:
I don't think we need this if statement, we always want to add computePrice to the list even if it is 0 in order to get an accourate sense of the percentile. The go code also defaults to 0 if it couldn't find a set price instruction: https://github.com/smartcontractkit/chainlink-solana/blob/priority-fee/pkg/solana/fees/utils.go#L73 in the current transaction